packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -132,8 +132,8 @@ export default util.createRule<Options, MessageIds>({
       requiresTypeChecking: true,
     },
     messages: {
-      unbound:
-        'Avoid referencing unbound methods which may cause unintentional scoping of `this`.',
+      unbound: `Avoid referencing unbound methods which may cause unintentional scoping of \`this\`.
+If you think it is intended, you can try to add \`this: void\` as the first function parameter or use arrow function instead.`,
```

Comment:
I don't think that we should have this in the message unless we know that the function does not have a `this` parameter.

eg it'd be weird if the user saw this message but the function was annotated with `this: Something`.

Reply:
OK, I'll add a check and new message for it.

Reply:
Done!